src/config.rs

```diff
@@ -32,34 +33,79 @@ impl FromStr for WorkMode {
 }
 
 #[derive(Debug)]
-pub enum ConfigError {
+pub enum Error {
+    CommandLine(DocoptError),
     Io(IoError, String),
-    TomlError(String, String),
+    Toml(String, String),
+    Options(String, String),
 }
 
-impl Error for ConfigError {
+impl StdError for Error {
     fn description(&self) -> &str {
+        use self::Error::*;
         match *self {
-            ConfigError::Io(ref e, _) => e.description(),
-            ConfigError::TomlError(ref s, _) => s,
+            CommandLine(ref e) => e.description(),
+            Io(ref e, _) => e.description(),
+            Toml(ref s, _) => s,
+            Options(ref s, _) => s,
         }
     }
 }
 
-impl Display for ConfigError {
+impl Display for Error {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        use self::Error::*;
         match *self {
-            ConfigError::Io(ref err, ref filename) => {
-                try!(write!(f, "\"{}\": ", filename));
+            CommandLine(ref err) => err.fmt(f),
+            Io(ref err, ref filename) => {
+                try!(write!(f, "Failed to read config \"{}\": ", filename));
                 err.fmt(f)
             }
-            ConfigError::TomlError(ref err, ref filename) => {
+            Toml(ref err, ref filename) => {
+                write!(f, "\"{}\": {}", filename, err)
+            }
+            Options(ref err, ref filename) => {
                 write!(f, "\"{}\": {}", filename, err)
             }
         }
     }
 }
 
+impl<'a> From<DocoptError> for Error {
+    fn from(e: DocoptError) -> Error {
+        Error::CommandLine(e)
+    }
+}
+
+impl<'a> From<(IoError, &'a str)> for Error {
+    fn from(e: (IoError, &'a str)) -> Error {
+        Error::Io(e.0, e.1.into())
+    }
+}
+
+impl<'a, 'b> From<(&'a str, &'b str)> for Error {
```

Comment:
Variance makes putting different lifetime parameters here unnecessary: a single `'a` will work out to be the shortest of the given borrows.